src/compiler/crystal/compiler.cr

```diff
@@ -80,14 +80,23 @@ module Crystal
     # the source file to compile.
     property prelude = "prelude"
 
-    # If `true`, runs LLVM optimizations.
-    property? release = false
+    # LLVM Optimization mode
+    enum OptimizationMode
+      O0
+      O1
+      O2
+      O3
+    end
```

Comment:
**suggestion:** Could you add some documentation what each modes means? The effects are a bit scattered through the code base so it's a good ideat to summarize. This definition is probably a good place to write that down as internal docs.

We'll also nee user documentation in the manual but that doesn't need to be part of this PR.